compose-app/src/main/java/com/mapbox/maps/extension/compose/viewport/MapViewport.kt

```diff
@@ -0,0 +1,85 @@
+package com.mapbox.maps.extension.compose.viewport
+
+import com.mapbox.maps.plugin.viewport.data.DefaultViewportTransitionOptions
+import com.mapbox.maps.plugin.viewport.data.FollowPuckViewportStateOptions
+import com.mapbox.maps.plugin.viewport.data.OverviewViewportStateOptions
+import java.util.*
+
+
+public sealed class MapViewport {
+  /**
+   * Represents the current status is a [FollowPuckState].
+   */
+  public class FollowPuckState(
+    /**
+     * The current [FollowPuckViewportStateOptions].
+     */
+    public val stateOptions: FollowPuckViewportStateOptions,
+    /**
```

Comment:
why do we have `FollowPuck` and `Overview` state here as well. can't we directly use them from viewport plugin?

Reply:
also these doesn't have any @composables.

Reply:
We can not use directly from ViewportPlugin as we would need MapboxMap to create the instances, while compose users don't have access to.